core/src/main/java/com/graphhopper/util/details/PathDetailsBuilderFactory.java

```diff
@@ -50,6 +54,20 @@
         if (requestedPathDetails.contains(DISTANCE))
             builders.add(new DistanceDetails());
 
+        for (String key : Arrays.asList(MaxSpeed.KEY, MaxWidth.KEY, MaxHeight.KEY, MaxWeight.KEY)) {
+            if (requestedPathDetails.contains(key) && encoder.hasEncodedValue(key))
+                builders.add(new DecimalDetails(key, encoder.getDecimalEncodedValue(key), true));
+        }
+
+        for (Map.Entry entry : Arrays.asList(new MapEntry<>(RoadClass.KEY, RoadClass.class),
```

Comment:
Could we extract these arrays to class level?

Reply:
What do you mean here?